sentry-core/src/main/java/io/sentry/core/SendFireAndForgetEnvelopeSender.java

```diff
@@ -0,0 +1,51 @@
+package io.sentry.core;
+
+import java.io.File;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+@ApiStatus.Internal
+public final class SendFireAndForgetEnvelopeSender
+    implements SendCachedEventFireAndForgetIntegration.SendFireAndForgetFactory {
+
+  private final @NotNull SendCachedEventFireAndForgetIntegration.SendFireAndForgetDirPath
+      sendFireAndForgetDirPath;
+
+  public SendFireAndForgetEnvelopeSender(
+      final @NotNull SendCachedEventFireAndForgetIntegration.SendFireAndForgetDirPath
+              sendFireAndForgetDirPath) {
+    this.sendFireAndForgetDirPath = sendFireAndForgetDirPath;
+  }
+
+  @Override
+  public SendCachedEventFireAndForgetIntegration.SendFireAndForget create(
+      final @NotNull IHub hub, final @NotNull SentryOptions options) {
+    final String dirPath = sendFireAndForgetDirPath.getDirPath();
+    if (dirPath == null) {
+      options
+          .getLogger()
+          .log(
+              SentryLevel.WARNING,
+              "No cache dir path is defined in options, discarding SendCachedEvent.");
```

Comment:
I'm a bit confused now because it implements `SendCachedEventFireAndForgetIntegration` which has `Event` in the name but this is targeting Envelopes.

```suggestion
              "No cache dir path is defined in options, discarding SendCachedEvelope.");
```

Down below it uses `final File sessionsPath = new File(dirPath);` so maybe it needs some renaming here

Reply:
yep, good catch, copy pasta typo